pipelines/imputation/scientificValidation/ReshapeReferencePanel.wdl

```diff
@@ -4,21 +4,499 @@ version 1.0
+    }
+    runtime {
+        docker: ubuntu_docker
+        disks: "local-disk ${disk_size_gb} HDD"
+        memory: "${memory_mb} MiB"
+        cpu: cpu
+        preemptible: 3
+    }
+    output {
+        Int chrom_length = read_int(stdout())
+    }
+}
+
+task GenerateChunk {
+    input {
+        Int start
+        Int end
+        String chrom
+        String basename
+        File vcf
+        File vcf_index
+
+        Int disk_size_gb = ceil(2*size(vcf, "GiB")) + 10
+        Int cpu = 1
+        Int memory_mb = 8000
+        String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.6.1.0"
+    }
+    Int command_mem = memory_mb - 1500
+    Int max_heap = memory_mb - 1000
+
+    command {
+        set -euo pipefail
+
+        gatk --java-options "-Xms~{command_mem}m -Xmx~{max_heap}m" \
+        SelectVariants \
+        -V ~{vcf} \
+        -L ~{chrom}:~{start}-~{end} \
+        -O ~{basename}.vcf.gz \
+        --exclude-filtered true
+    }
+    runtime {
+        docker: gatk_docker
+        disks: "local-disk ${disk_size_gb} HDD"
+        memory: "${memory_mb} MiB"
+        cpu: cpu
+    }
+    parameter_meta {
+        vcf: {
+                 description: "vcf",
+                 localization_optional: true
+             }
+        vcf_index: {
+                       description: "vcf index",
+                       localization_optional: true
+                   }
+    }
+    output {
+        File output_vcf = "~{basename}.vcf.gz"
+        File output_vcf_index = "~{basename}.vcf.gz.tbi"
+    }
+}
+
+task GatherVcfs {
+    input {
+        Array[File] input_vcfs
+        String output_vcf_name
+        Int disk_size_gb = ceil(1.5 * size(input_vcfs, "GiB")) + 10
+        Int machine_mem_mb = 6000
+        String gatk_docker = "us.gcr.io/broad-gatk/gatk:4.6.1.0"
+    }
+
+    parameter_meta {
+        input_vcfs: {
+                        localization_optional: true
+                    }
+    }
+
+    command <<<
+        set -euo pipefail
+
+        # --ignore-safety-checks makes a big performance difference so we include it in our invocation.
+        # This argument disables expensive checks that the file headers contain the same set of
+        # genotyped samples and that files are in order by position of first record.
+        gatk --java-options "-Xms4000m -Xmx4500m" \
+        GatherVcfsCloud \
+        --ignore-safety-checks \
+        --gather-type BLOCK \
+        --input ~{sep=" --input " input_vcfs} \
+        --output ~{output_vcf_name}
+
+        tabix ~{output_vcf_name}
+    >>>
+
+    runtime {
+        memory: "~{machine_mem_mb} MiB"
+        cpu: "1"
+        bootDiskSizeGb: 15
+        disks: "local-disk " + disk_size_gb + " HDD"
+        preemptible: 0
+        docker: gatk_docker
+    }
+
+    output {
+        File output_vcf = "~{output_vcf_name}"
+        File output_vcf_index = "~{output_vcf_name}.tbi"
+    }
+}
+
+task UpdateHeader {
+    input {
+        File vcf
+        File vcf_index
+        File ref_dict
```

Comment:
Possible this could be localization_optional (many gatk tools allow this now). But not sure.

Reply:
i can try it